frontend/src/components/App/Home/index.tsx

```diff
@@ -236,6 +236,24 @@ function HomeComponent(props: HomeComponentProps) {
       .sort();
   }
 
+  /**
+   * Gets the origin of a cluster.
+   *
+   * @param cluster
+   * @returns A description of where the cluster is picked up from: dynamic, in-cluster, or from a kubeconfig file.
+   */
+  function getOrigin(cluster: Cluster): string {
+    if (cluster.meta_data?.source === 'kubeconfig') {
+      const kubeconfigPath = process.env.KUBECONFIG ?? '~/.kube/config';
+      return t('translation|Kubeconfig') + ` (${kubeconfigPath})`;
+    } else if (cluster.meta_data?.source === 'dynamic_cluster') {
+      return t('translation|Dynamic cluster');
```

Comment:
I am not sure this will give a lot of info to the user. I'd prefer the still poor but more accurate `Plugin` for now.

Reply:
Got it